### PR TITLE
feat(connection): load connection data on extension startup + edit/remove/switch connections

### LINKED ISSUE #171: Can't Edit or Remove connections
I'm using vsflux in VS Code 1.50.0 under macOS Catalina.

Add Connection worked fine to add two connections, but now I need to delete one of them, and find that both Edit Connection and Remove Connection result in errors like this one: 

`Command 'InfluxDB: Edit Connection' resulted in an error (Cannot read property 'edit' of undefined)`

I've un- and re-installed the plugin to no avail.

Can anyone give me a hint on where to look to manually delete one or both?

And is there additional information I might be able to collect to help figure out why Edit and Remove aren't working?

EDIT:
When I restart VS Code, I have to Add Connection again--so, it appears that either the connections I create are never actually saved (which could explain Edit and Remove not working), or at least that no default connection is remembered.

### PR DESCRIPTION
**Pull request by scbrickley**
Closes #171 

This PR does a few things:

Major changes:
- Exposes `switchConnection` action to the user
- Loads last active connection on extension startup
- Fixes bug where the `ConnectionNode` being passed into `editConnection`, `removeConnection`, and `switchConnection` was undefined.

Minor changes:
- Changes default URL for an influxdb connection from `localhost:9999` to `localhost:8086` to reflect change in OSS
- Renames `InfluxDBConectionsKey` to `InfluxDBConnectionsKey`
- makes `tree.getConnectionNodes` public

Concerns I have:
- `switch`, `edit`, and `removeConnection` actions won't work properly until the extension is started (either by adding a connection or loading a `.flux` file)
- Some of the indentation is inconsistent, so I'd like to reformat using tabs rather than spaces, or at least run a typescript formatting tool, but I'm holding off on that, because that would make the diff hard to read.
- Most of the variable declarations in the repo used `const` rather than `var` or `let`, and I don't know enough about typescript to know what best practice is. `var` seems to be the standard, and that's what I've used in most of the added code, but I wonder if there's a reason that the original author decided to use `const`.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export async function activate (context: ExtensionContext) {
   Connection.load(context)
 
   client = new Client(context)
-  await client.start()
+  client.start()
```
Comment (rockstar) [inline src/extension.ts:17]: I don't see a change to `client.start` that would make `await`ing it unnecessary. Can you help me understand why this change was made?

**Review by rockstar — DISMISSED** [review]
These changes all look good. Thanks for chasing it.

> Some of the indentation is inconsistent, so I'd like to reformat using tabs rather than spaces, or at least run a typescript formatting tool, but I'm holding off on that, because that would make the diff hard to read.

I'd suggest using a formatting tool, and taking whatever it gives you. There was a time when Crockford said "Tabs are evil, use spaces" but it seems that the javascript (and, in effect, typescript) community is re-hashing this one.

> Most of the variable declarations in the repo used const rather than var or let, and I don't know enough about typescript to know what best practice is. var seems to be the standard, and that's what I've used in most of the added code, but I wonder if there's a reason that the original author decided to use const.

The convention that I have seen, followed, and agreed with is that one should default to `const` as the variable would be immutable-ish (ugh) and the scoping would stay in the block. For mutability, use `let`, as the scope then stays block-local. `var` seems deprecated-by-convention more than anything.

File: src/extension.ts (hunk 1 above)
Comment (scbrickley) [reply src/extension.ts:17]: That was just me appeasing the typescript linter. Message said `'await' has no effect on this type of expression`

Comment (rockstar) [reply src/extension.ts:17]: Ah, so it didn't need to be `await`ed in the first place. Gotcha.

**Comment by scbrickley** [comment]
Thanks @rockstar! I'll make the changes to the variable declarations.

**Review by rockstar — APPROVED** [review]
